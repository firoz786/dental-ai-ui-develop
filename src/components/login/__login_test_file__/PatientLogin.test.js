import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import PatientLogin from "../Login";
import { BrowserRouter as Router } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
const LoginPage = () => {
  return (
    <Router>
      <PatientLogin />
    </Router>
  );
};

jest.setTimeout(30000);

describe("Writing test case for Patient Login Page", () => {
  window.matchMedia =
    window.matchMedia ||
    function () {
      return {
        matches: false,
        addListener: function () {},
        removeListener: function () {},
      };
    };
  test("testing for PatientLogin page rendering or not", () => {
    render(<LoginPage />);
  });
  test("1, testing  for 'Ivory-logo' appearing or not", () => {
    render(<LoginPage />);
    const ivoryLogo = screen.getByTestId("Ivory-logo");
    expect(ivoryLogo).toBeInTheDocument();
  });
  test("2, testing for 'Login to' text rendering or not", () => {
    render(<LoginPage />);
    const LoginToText = screen.getByText("Login to");
    expect(LoginToText).toBeInTheDocument();
  });
  test("3, testing for 'your Account' text rendering or not", () => {
    render(<LoginPage />);
    const yourAccount = screen.getByText("your Account");
    expect(yourAccount).toBeInTheDocument();
  });
  test("4, testing for 'Username' text rendering or not", () => {
    render(<LoginPage />);
    const UserName = screen.getByText("Username");
    expect(UserName).toBeInTheDocument();
  });
  test("5, testing for 'user-name-input' input tag appearing or not", () => {
    render(<LoginPage />);
    const userNameInput = screen.getByTestId("user-name-input");
    expect(userNameInput).toBeInTheDocument();
  });
  test("6, testing for 'Password' text rendering or not", () => {
    render(<LoginPage />);
    const Password = screen.getByText("Password");
    expect(Password).toBeInTheDocument();
  });
  test("7, testing for 'password-input' input tag appearing or not", () => {
    render(<LoginPage />);
    const passwordInput = screen.getByTestId("password-input");
    expect(passwordInput).toBeInTheDocument();
  });
  test("8, testing for 'Forgot Password' text rendering or not", () => {
    render(<LoginPage />);
    const ForgotPassword = screen.getByText("Forgot Password");
    expect(ForgotPassword).toBeInTheDocument();
  });
  test("9, test case for 'remember-me-check-box' check-box is appearing or not", () => {
    render(<LoginPage />);
    const rememberMeCheckbox = screen.getByTestId("remember-me-check-box");
    expect(rememberMeCheckbox).toBeInTheDocument();
  });
  test("10, testing for 'Remember me' text rendering or not", () => {
    render(<LoginPage />);
    const rememberMe = screen.getByText("Remember me");
    expect(rememberMe).toBeInTheDocument();
  });
  test("11, test case for 'login-submit-btn' is appearing or not", () => {
    render(<LoginPage />);
    const loginSubmitBtn = screen.getByTestId("login-submit-btn");
    expect(loginSubmitBtn).toBeInTheDocument();
  });
  test("12, test case for 'Login' text on loginBtn rendering or not", () => {
    render(<LoginPage />);
    const login = screen.getByText("Login");
    expect(login).toBeInTheDocument();
  });
  //   test("13 test case for 'Copy rights' text rendering or not", () => {
  //     render(<LoginPage />);
  //     const copyRightsText = screen.getByText(
  //       "Copyright @ 2023. All Rights Reserved"
  //     );
  //     expect(copyRightsText).toBeInTheDocument();
  //   });
  test("14, test case for 'New Here?' text rendering or not", () => {
    render(<LoginPage />);
    const newHere = screen.getByText("New Here?");
    expect(newHere).toBeInTheDocument();
  });
  test("15, test case for 'suggestion text in login page' text rendering or not", () => {
    render(<LoginPage />);
    const suggestionText = screen.getByText(
      "Sign up and discover a whole new World of AI based treatment"
    );
    expect(suggestionText).toBeInTheDocument();
  });
  test("16, test case for 'sign-up-btn' in login page is appearing or not", () => {
    render(<LoginPage />);
    const signUpBtn = screen.getByTestId("sign-up-btn");
    expect(signUpBtn).toBeInTheDocument();
  });
  test("17, test case for 'Sign Up' text on Sign-up-Btn in login page is rendering or not", () => {
    render(<LoginPage />);
    const signUp = screen.getByText("Sign up");
    expect(signUp).toBeInTheDocument();
  });
  test("18, test case to username for error message", async () => {
    render(<LoginPage />);
    const loginSubmitBtn = screen.getByTestId("login-submit-btn");
    fireEvent.click(loginSubmitBtn);
    await new Promise((r) => setTimeout(r, 1000));
    const enterUserName = screen.getByText("Please enter your username!");
    expect(enterUserName).toBeInTheDocument();
  });
  test("19, test case to password for error message", async () => {
    render(<LoginPage />);
    const loginSubmitBtn = screen.getByTestId("login-submit-btn");
    fireEvent.click(loginSubmitBtn);
    await new Promise((r) => setTimeout(r, 1000));
    const enterPassword = screen.getByText("Please enter your password!");
    expect(enterPassword).toBeInTheDocument();
  });
  test("20, test case for error message when user enter less characters in userName Input", async () => {
    render(<LoginPage />);
    const userNameInput = screen.getByTestId("user-name-input");
    userEvent.type(userNameInput, "Bh");
    await new Promise((r) => setTimeout(r, 1000));
    const enterMinChar = screen.getByText("Please enter minimum 4 characters");
    expect(enterMinChar).toBeInTheDocument();
  });
  test("21, test case for error message when user enter more than 30 characters in userName Input", async () => {
    render(<LoginPage />);
    const userNameInput = screen.getByTestId("user-name-input");
    userEvent.type(userNameInput, "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb");
    await new Promise((r) => setTimeout(r, 1000));
    const enterMaxChar = screen.getByText("Please enter miximum 30 characters");
    expect(enterMaxChar).toBeInTheDocument();
  });
  test("22, test case for error message when user enter only white spaces in userName Input", async () => {
    render(<LoginPage />);
    const userNameInput = screen.getByTestId("user-name-input");
    userEvent.type(userNameInput, "      ");
    await new Promise((r) => setTimeout(r, 1000));
    const whiteSpaces = screen.getByText("No white spaces");
    expect(whiteSpaces).toBeInTheDocument();
  });
  test("23, test case for error message when user enter only white spaces and enter less than 4 characters in userName Input", async () => {
    render(<LoginPage />);
    const userNameInput = screen.getByTestId("user-name-input");
    userEvent.type(userNameInput, "  ");
    await new Promise((r) => setTimeout(r, 1000));
    const whiteSpaces = screen.getByText("No white spaces");
    expect(whiteSpaces).toBeInTheDocument();
    const enterMinChar = screen.getByText("Please enter minimum 4 characters");
    expect(enterMinChar).toBeInTheDocument();
  });
  test("24, test case for error message when user only enter white spaces and enter more than 30 characters", async () => {
    render(<LoginPage />);
    const userNameInput = screen.getByTestId("user-name-input");
    userEvent.type(userNameInput, "                                ");
    await new Promise((r) => setTimeout(r, 1000));
    const whiteSpaces = screen.getByText("No white spaces");
    expect(whiteSpaces).toBeInTheDocument();
    const enterMaxChar = screen.getByText("Please enter miximum 30 characters");
    expect(enterMaxChar).toBeInTheDocument();
  });
  test("25, test case for error message when user enter wrong pattern in Password Input", async () => {
    render(<LoginPage />);
    const passwordInput = screen.getByTestId("password-input");
    userEvent.type(passwordInput, "bhaf");
    await new Promise((r) => setTimeout(r, 1000));
    const passwordPattern = screen.getByText(
      "Password must contain a capital letter and at least one alphanumeric character and one symbol from @#$%^&*_"
    );
    expect(passwordPattern).toBeInTheDocument();
  });
  test("26, test case for error message when user enter only white spaces in Password Input", async () => {
    render(<LoginPage />);
    const passwordInput = screen.getByTestId("password-input");
    userEvent.type(passwordInput, "  ");
    await new Promise((r) => setTimeout(r, 1000));
    const whiteSpaces = screen.getByText("No white spaces");
    expect(whiteSpaces).toBeInTheDocument();
    const passwordPattern = screen.getByText(
      "Password must contain a capital letter and at least one alphanumeric character and one symbol from @#$%^&*_"
    );
    expect(passwordPattern).toBeInTheDocument();
  });
  test("27, test case for error message when user enter only white spaces more than 30 characters in Password Input", async () => {
    render(<LoginPage />);
    const passwordInput = screen.getByTestId("password-input");
    userEvent.type(passwordInput, "                                ");
    await new Promise((r) => setTimeout(r, 1000));
    const enterMaxChar = screen.getByText(
      "Please enter miximum 30 characters only"
    );
    expect(enterMaxChar).toBeInTheDocument();
    const whiteSpaces = screen.getByText("No white spaces");
    expect(whiteSpaces).toBeInTheDocument();
    const passwordPattern = screen.getByText(
      "Password must contain a capital letter and at least one alphanumeric character and one symbol from @#$%^&*_"
    );
    expect(passwordPattern).toBeInTheDocument();
  });
  test("28, test case for error message when user enter more than 30 characters and without a perfect password pattern in Password Input", async () => {
    render(<LoginPage />);
    const passwordInput = screen.getByTestId("password-input");
    userEvent.type(passwordInput, "                                ");
    await new Promise((r) => setTimeout(r, 1000));
    const enterMaxChar = screen.getByText(
      "Please enter miximum 30 characters only"
    );
    expect(enterMaxChar).toBeInTheDocument();
    const passwordPattern = screen.getByText(
      "Password must contain a capital letter and at least one alphanumeric character and one symbol from @#$%^&*_"
    );
    expect(passwordPattern).toBeInTheDocument();
  });
  test("29, test case for error message when user enter more than 30 characters and with proper password pattern in Password Input", async () => {
    render(<LoginPage />);
    const passwordInput = screen.getByTestId("password-input");
    userEvent.type(
      passwordInput,
      "Bhargav@12bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
    );
    await new Promise((r) => setTimeout(r, 1000));
    const enterMaxChar = screen.getByText(
      "Please enter miximum 30 characters only"
    );
    expect(enterMaxChar).toBeInTheDocument();
  });
  test("30, test case for error message when user entered correct credentials", async () => {
    render(<LoginPage />);
    const userNameInput = screen.getByTestId("user-name-input");
    userEvent.type(userNameInput, "Bhargav");
    const passwordInput = screen.getByTestId("password-input");
    userEvent.type(passwordInput, "Bhargav@12");
    const loginSubmitBtn = screen.getByTestId("login-submit-btn");
    fireEvent.click(loginSubmitBtn);
    await new Promise((r) => setTimeout(r, 2000));
    const wrongCredetials = screen.getByText("Successfully Logged in !");
    expect(wrongCredetials).toBeInTheDocument();
  });
  test("31, test case for error message when user entered wrong credentials", async () => {
    render(<LoginPage />);
    const userNameInput = screen.getByTestId("user-name-input");
    userEvent.type(userNameInput, "Bhargavvvv");
    const passwordInput = screen.getByTestId("password-input");
    userEvent.type(passwordInput, "Bhargav@12");
    const loginSubmitBtn = screen.getByTestId("login-submit-btn");
    fireEvent.click(loginSubmitBtn);
    await new Promise((r) => setTimeout(r, 2000));
    const wrongCredetials = screen.getByText(
      "Invalid username or password. Please try again."
    );
    expect(wrongCredetials).toBeInTheDocument();
  });
  test("Snapshot testing for PatientLoginPage", () => {
    const rendderingPatientLoginPage = renderer.create(<LoginPage />).toJSON();
    expect(rendderingPatientLoginPage).toMatchSnapshot();
  });
});
