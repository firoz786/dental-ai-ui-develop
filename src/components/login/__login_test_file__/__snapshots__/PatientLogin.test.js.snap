// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Writing test case for Patient Login Page Snapshot testing for PatientLoginPage 1`] = `
<div
  className="ant-row login-page-main-row"
  style={Object {}}
>
  <div
    className="ant-col ant-col-16 login-details-filling-form-col-tag ant-col-xs-24 ant-col-sm-16"
    style={Object {}}
  >
    <div
      className="ant-row"
      style={
        Object {
          "height": "10vh",
        }
      }
    >
      <img
        alt=""
        className="Ivory-logo-img-tag"
        data-testid="Ivory-logo"
        src="ivory-ai-logo.png"
      />
    </div>
    <div
      className="ant-row login-details-filling-row-tag"
      style={Object {}}
    >
      <form
        autoComplete="off"
        className="ant-form ant-form-horizontal login-details-filling-form"
        id="basic"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="ant-row login-to-youracnt-row-text"
          style={Object {}}
        >
          <span
            className="loginto-text-span-tag"
          >
            Login to
          </span>
          <span
            className="loginto-text-span-tag-two"
          >
            your Account
          </span>
        </div>
        <span
          className="username-text-span-tag"
        >
          Username
        </span>
        <div
          className="ant-form-item"
        >
          <div
            className="ant-row ant-form-item-row"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <input
                    aria-required="true"
                    className="ant-input patient-signin-input"
                    data-testid="user-name-input"
                    id="basic_name"
                    label="name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          className="password-text-span-tag"
        >
          Password
        </span>
        <div
          className="ant-form-item"
        >
          <div
            className="ant-row ant-form-item-row"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <span
                    className="ant-input-affix-wrapper ant-input-password patient-signin-input"
                    onClick={[Function]}
                  >
                    <input
                      aria-required="true"
                      className="ant-input"
                      data-testid="password-input"
                      hidden={null}
                      id="basic_password"
                      label="password"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      style={null}
                      type="password"
                      value=""
                    />
                    <span
                      className="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        className="anticon anticon-eye-invisible ant-input-password-icon"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseUp={[Function]}
                        role="img"
                        tabIndex={-1}
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-form-item"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="ant-row ant-form-item-row"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <div
                    checked={true}
                    className="ant-row"
                    id="basic_remember"
                    onChange={[Function]}
                    style={
                      Object {
                        "display": "flex",
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      className="ant-col ant-col-12"
                      style={Object {}}
                    >
                      <span
                        className="forgot-password-span-tag"
                      >
                        <u>
                          Forgot Password
                        </u>
                      </span>
                    </div>
                    <div
                      className="ant-col ant-col-12 rememberme-checkbox-col-tag"
                      style={Object {}}
                    >
                      <label
                        className="ant-checkbox-wrapper ant-checkbox-wrapper-in-form-item remeber-me-checkbox-tag"
                      >
                        <span
                          className="ant-checkbox"
                          style={Object {}}
                        >
                          <input
                            checked={false}
                            className="ant-checkbox-input"
                            data-testid="remember-me-check-box"
                            disabled={false}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyPress={[Function]}
                            onKeyUp={[Function]}
                            type="checkbox"
                          />
                          <span
                            className="ant-checkbox-inner"
                          />
                        </span>
                        <span>
                          Remember me
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-form-item login-btn-form-item-tag"
        >
          <div
            className="ant-row ant-form-item-row"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <button
                    className="ant-btn ant-btn-primary login-button-tag"
                    data-testid="login-submit-btn"
                    disabled={false}
                    onClick={[Function]}
                    type="submit"
                  >
                    <span>
                      Login
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      className="ant-row copy-rights-text-row-tag"
      style={Object {}}
    >
      <span
        className="copy-rights-text-span-tag"
      >
        Copyright @ 2023. All Rights Reserved
      </span>
    </div>
  </div>
  <div
    className="ant-col ant-col-8 signup-suggestion-img-col ant-col-xs-24 ant-col-sm-8"
    style={Object {}}
  >
    <div
      className="ant-row signup-text-row-tag"
      style={Object {}}
    >
      <span
        className="new-here-text-span-row"
      >
        New Here?
      </span>
      <span
        className="sign-up-text-span-tag"
      >
        Sign up and discover a whole new World of AI based treatment
      </span>
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      <button
        className="ant-btn ant-btn-default sign-up-btn-login-page"
        data-testid="sign-up-btn"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          Sign up
        </span>
      </button>
    </div>
  </div>
  <div
    className="ant-col copy-rights-text"
    style={Object {}}
  >
    <span
      className="copy-rights-text-span"
    >
      Copyright @ 2023. All Rights Reserved
    </span>
  </div>
</div>
`;
